### DISCUSSION TITLE
no screen TFT24 V1.1 to board SKR2 B (F429) connection

### DISCUSSION THREAD

**Discussion by ANTONIVA** [bigtreetech/BIGTREETECH-TouchScreenFirmware#2513] [Q&A] [answered]
I need your help!
Motherboard SKR2 V B STM32f429 and BIGTREE TFT24 V1.1 no connection! Сonnection works only on EXP1 and EXP2!
In firmware Marlin bugfix-2.0.9.2.x the following settings are available:
configuration.h
#define SERIAL_PORT -1
#define BAUDRATE 250000
#define SERIAL_PORT_2 1
(at values SERIAL_PORT_2 0 - writes an error!)
In firmware BIGTREETECH-TouchScreenFirmware-master the following settings are available:
#define SP_1 7  //"Options: [OFF (port disabled): 0, 2400: 1, 9600: 2, 19200: 3, 38400: 4, 57600: 5, 115200: 6, 250000: 7, 500000: 8, 1000000: 9]"
The cables to the device are connected correctly!
what am I doing wrong?
Sorry for my english.

**Comment by kisslorand** [comment]
You should set the proper baud rate for the TFT in the "config.ini", not in the "Configuration.h".

```
#   Options: [OFF (port disabled): 0, 2400: 1, 9600: 2, 19200: 3, 38400: 4, 57600: 5, 115200: 6, 250000: 7, 500000: 8, 1000000: 9]
serial_port:P1:7 P2:7 P3:0 P4:0
```

...or you can do it on the TFT:  Menu -> Settings -> Connection -> S. Ports

**Reply by ANTONIVA** [reply]
![IMG_20220502_104046](https://user-images.githubusercontent.com/104736492/166201560-d428f28c-dec0-4d5a-a304-a784f54d6fdb.jpg)
![IMG_20220502_103729](https://user-images.githubusercontent.com/104736492/166201593-e3979cdd-9e1f-4ab4-adaa-40ff08094c7b.jpg)
![IMG_20220502_103754](https://user-images.githubusercontent.com/104736492/166201610-fbda4a7a-a3eb-46c3-a393-dfa872cdbf82.jpg)

**Reply by ANTONIVA** [reply]
![IMG_20220502_104027](https://user-images.githubusercontent.com/104736492/166201729-7ef44dce-5042-4ccc-bf10-fa143447f541.jpg)
![IMG_20220502_103935](https://user-images.githubusercontent.com/104736492/166201736-782903d3-0dcb-495b-9952-9b5a7135a8d3.jpg)

**Reply by ANTONIVA** [reply]
Thanks for helping me with my problem!
The BAUDRATE is the same on the board and on the TFT. And there is no connection TFT :(

**Reply by ANTONIVA** [reply]
I checked the correspondence of the pins on the boards. I checked for breaks on the board and on the wire. Everything is okay.

**Comment by artineo** [comment]
it's putin's fault

**Reply by ANTONIVA** [reply]
I live in Ukraine. I hope this person gets into the Gaga Court. He and his generals. 
Ukrainians speak two languages ​​fluently. Я вільно розмовляю україньською та російською мовами. Ми вільна держава, ми вільний народ!
Unfortunately, I could not solve the problem with the screen. I think it's firmware SKR 2B.

**Comment by kisslorand** [comment, marked as answer]
I have the same motherboard as you, I can send you the firmware I use so you can test it if it's FW problem or not.

Слава Україні!

**Reply by ANTONIVA** [reply]
Героям слава!
I'll be very grateful!
If will work!

**Comment by artineo** [comment]
I have SKR mini E3. My settings
![20220519_153718](https://user-images.githubusercontent.com/63225270/169307932-94a2771e-6ffa-482b-94bf-614c3abecee0.jpg)
![20220519_153805](https://user-images.githubusercontent.com/63225270/169307965-131356db-060b-49cc-b3ba-b9aa16892d51.jpg)
:

Do You check all this settings in marlin? :


General options which MUST be activated:

EEPROM_SETTINGS (in Configuration.h)
BABYSTEPPING (in Configuration_adv.h)
AUTO_REPORT_TEMPERATURES (in Configuration_adv.h)
AUTO_REPORT_POSITION (in Configuration_adv.h)
M115_GEOMETRY_REPORT (in Configuration_adv.h)
M114_DETAIL (in Configuration_adv.h)
REPORT_FAN_CHANGE (in Configuration_adv.h)

Options to support printing from onboard media:

SDSUPPORT (in Configuration.h)
LONG_FILENAME_HOST_SUPPORT (in Configuration_adv.h)
AUTO_REPORT_SD_STATUS (in Configuration_adv.h)
SDCARD_CONNECTION ONBOARD (in Configuration_adv.h)

Options to support dialog with host:

EMERGENCY_PARSER (in Configuration_adv.h)
SERIAL_FLOAT_PRECISION 4 (in Configuration_adv.h)
HOST_ACTION_COMMANDS (in Configuration_adv.h)
HOST_PROMPT_SUPPORT (in Configuration_adv.h)

Options to support M600 with host & (Un)Load menu:

Options to support dialog with host as pre requisite
NOZZLE_PARK_FEATURE (in Configuration.h)
ADVANCED_PAUSE_FEATURE (in Configuration_adv.h)
PARK_HEAD_ON_PAUSE (in Configuration_adv.h)
FILAMENT_LOAD_UNLOAD_GCODES (in Configuration_adv.h)

Options to fully support Bed Leveling menu:

Z_MIN_PROBE_REPEATABILITY_TEST (in Configuration.h)
G26_MESH_VALIDATION (in Configuration.h)
Z_STEPPER_AUTO_ALIGN (in Configuration_adv.h)

**Reply by ANTONIVA** [reply]
Thank you! Check my firmware

**Comment by kisslorand** [comment]
Here's my Marlin FW for SKR v2 with STM32F429.
[FIRMWARE](https://drive.google.com/file/d/1yAIDXpwmQrM7uqs1abp5mCSZrop5YKVw/view?usp=sharing)

**Reply by ANTONIVA** [reply]
Дякую друже! Перевірю, дуже вдячний за підтримку! :)

**Comment by ANTONIVA** [comment]
Thanks guys!
Today I will test the firmware! Look for your mistakes :) As soon as I get the result, I will write a report.

Sorry for not showing up. Went for family and brought them home.

**Comment by radek8** [comment]
You are not alone with this problem.
The most common problem is a faulty serial cable, faulty configuration, or faulty pins on the TFT or SKR board.
Check where you connected the serial cable. Some users mistakenly used the SWD connector !!
If you have a good serial cable, a properly activated serial port in Merlina and the same baud rate, then you have a problem with a faulty pin on the processor.
Look here.

https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/issues/2382
https://github.com/bigtreetech/BIGTREETECH-TFT24-V1.1/issues/35
https://github.com/bigtreetech/BIGTREETECH-TFT35-V3.0/issues/120
